### DISCUSSION TITLE
HomeKit - Adaptive Lighting

### DISCUSSION THREAD

**Discussion by ol1bergmann** [brutella/hkknx-public#164] [Ideas]
Hallo Brutella,

Ist es geplant die Funktion "Adaptive Lighting" zu integrieren? 

Ich habe einige CW WW LEDs (DALI KNX Gateway, Lunatone LED Treiber) im Einsatz und würde mich freuen wenn es die Funktionalität "Adaptive Lightning" in Zukunft geben würde.

![image](https://user-images.githubusercontent.com/25728214/151130347-ec75e858-053a-4449-9c86-42d6ddbac444.png)

**Comment by brutella** [comment]
Ich bin gerade dabei die Unterstützung für Adaptive Lighting zu implementieren. 

Neben HomeKit könnte man diese Funktion auch über Skripte lösen. 

HomeKit hat aber den Vorteil, dass man über die Apple Home-App die Funktion für einzelne Lampen relativ einfach temporär deaktivieren kann. Da stellt sich für mich aber die Frage, ob man das überhaupt braucht. Also wenn man einmal die Funktion aktiviert hat, lässt man es dann nicht ohnehin immer eingeschaltet?

@ol1bergmann Wie siehst du das?

**Reply by edelmaca** [reply]
Ich würde mich sehr über eine Umsetzung hier freuen.

[a comment by an automated account is not shown]

**Comment by ol1bergmann** [comment]
Hi @brutella 
ich sehe es so wie du es bereits beschrieben hast. 
Tendenziell lässt man es sicherlich eingeschaltet, es sei denn man möchte die Lichtfarbe (CW/WW) am Abend doch etwas "lebendiger / frischer" (also mehr CW Anteil) haben. Das kann man dann ja komfortabel über die Home-App steuern.

Aber schön zu hören das du da dran bist und es die Funktion geben wird.

**Comment by mbrockeu** [comment]
Wollte mal Fragen wie hier der Stand ist, ist das noch auf der Roadmap oder runtergeflogen?
Grundsätzlich ist bei meinen Aktoren HCL aktiviert. Aber wenn man es übersteuert wäre es natürlich schön wenn man das über die Home App auch wieder anschalten kann.
Kann man zwar auch über einen Schalter umsetzen wäre aber natürlich viel schöner in der Lampe :-)

**Comment by brutella** [comment]
Bei Adaptive Lighting in HomeKit wird die Farbtemperatur über HomeKit gesteuert und hat so gesehen nichts mit HCL der KNX Aktoren zu tun. Verstehe ich das richtig, dass du das HCL der KNX Aktoren über HomeKit ein- und ausschalten möchtest?

**Reply by mbrockeu** [reply]
Hätte dort jetzt prinzipiell als das gleiche Verstanden. Zumindest das Ergebnis sollte ja ähnlich sein.

In den Aktoren kann man ja zumindest über eine GA die HCL Sequenz starten, z.b. falls sie mal überschrieben wurde.
Das ich jetzt über einen Schalter und Status das aktivieren kann ist mir klar. Aber hätte jetzt gedacht, wenn man auf dieses Adaptive Lighting Button drückt und dann z.b. die GA für HCL gestartet wird. 

Oder sagt der HomeKit Hub für die richtige Farbmischung?

**Comment by brutella** [comment]
- HCL in KNX wird vom Aktor aus betrieben. 
- HCL (Adaptive Lighting) geht von der HomeKit Steuerzentrale aus. Genauer gesagt sendet HomeKit periodisch an jede HCL konfigurierte Lampe die Farbtemperaturen für die jeweiligen Uhrzeiten. Die Lampe stellt dann je nach Uhrzeit die Farbtemperatur ein.

**Reply by mbrockeu** [reply]
Ah ok. Ja gut das wird dann nicht so funktionieren wie mein Ansatz. Dann braucht man HCL vermutlich nicht unbedingt, also ich dann wohl eher nicht. 

Man kann ja in HomeKit sicher auch nicht einstellen zu welcher Zeit was genau genommen werden soll im Gegensatz zum Aktor.

**Comment by brutella** [comment]
Genau, HomeKit hat eine eigene Vorstellung welche Farbtemperatur zu welcher Uhrzeit am Besten ist.

Adaptive Lighting macht Sinn, wenn man HCL in KNX haben möchte aber keinen passenden KNX Aktor dafür hat.

**Comment by thorsten-gehrig** [comment]
Ist denn Adaptive Lighting mittlerweile möglich? Wenn ja: wie?

**Reply by brutella** [reply]
Adaptive Lighting ist (noch) nicht verfügbar.

**Reply by edelmaca** [reply]
Also noch in der mache? 😉  Wäre richtig toll wenn das noch kommen würde.

**Comment by Zirkelino** [comment]
Gibt es ein Update zu dem Feature?
